pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -410,15 +410,17 @@ func groupPods(pods []*v1.Pod, metrics metricsclient.PodMetricsInfo, resource v1
 	return
 }
 
-func calculatePodRequests(pods []*v1.Pod, resource v1.ResourceName) (map[string]int64, error) {
+func calculatePodRequests(pods []*v1.Pod, container string, resource v1.ResourceName) (map[string]int64, error) {
 	requests := make(map[string]int64, len(pods))
 	for _, pod := range pods {
 		podSum := int64(0)
-		for _, container := range pod.Spec.Containers {
-			if containerRequest, ok := container.Resources.Requests[resource]; ok {
-				podSum += containerRequest.MilliValue()
-			} else {
-				return nil, fmt.Errorf("missing request for %s", resource)
+		for _, c := range pod.Spec.Containers {
+			if container == "" || container == c.Name {
+				if containerRequest, ok := c.Resources.Requests[resource]; ok {
+					podSum += containerRequest.MilliValue()
+				} else {
+					return nil, fmt.Errorf("missing request for %s", resource)
+				}
 			}
 		}
```

Comment:
As mentioned above (https://github.com/kubernetes/kubernetes/pull/90691#discussion_r463440392):

If a container is missing entirely, this will return a resource request of zero.  This is not ideal.  If the targeted container is missing, we should treat it as an invalid configuration and return an error.

Reply:
@josephburnett I've updated the logic to return an error in case the pod metrics endpoint does not return metrics about the the container.